src/utilities/byProgram.js

```diff
@@ -30,6 +33,13 @@ const getAllProgramsForCustomer = (customer, database) => {
   return database
     .objects('MasterListNameJoin')
     .filtered('name.id = $0 && masterList.isProgram = $1', id, true)
+    .filter(
+      ({ masterList }) =>
+        database
+          .objects('MasterListNameJoin')
+          .filtered('masterList.id == $0 && name.id == $1', masterList.id, thisStoresNameID)
```

Comment:
If make above suggested change:

```suggestion
          .filtered('masterList.id == $0 && name.id == $1', masterList.id, thisStoreNameID)
```